### PR TITLE
Move linting to ruff

### PR DESCRIPTION
**Pull request by adamltyson**
Copying the [changes made to movement](https://github.com/neuroinformatics-unit/movement/pull/156).

I have disabled the `pydocstyle` checks as there were too many issues raised.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,32 +6,55 @@ repos:
...
           - id: check-toml
+          - id: debug-statements
           - id: end-of-file-fixer
           - id: mixed-line-ending
             args: [--fix=lf]
+          - id: name-tests-test
+            args: ["--pytest-test-first"]
           - id: requirements-txt-fixer
           - id: trailing-whitespace
+    - repo: https://github.com/pre-commit/pygrep-hooks
```
Comment (niksirbi) [inline .pre-commit-config.yaml:25]: These hooks can be removed because they are for rst

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -6,32 +6,55 @@ repos:
...
-          - id: black
+        - id: ruff-format
     - repo: https://github.com/pre-commit/mirrors-mypy
       rev: v1.9.0
       hooks:
           - id: mypy
             additional_dependencies:
+                - attrs
                 - types-setuptools
+                - pandas-stubs
```
Comment (niksirbi) [inline .pre-commit-config.yaml:43]: Remove unnecessary stubs

File: pyproject.toml [hunk 3]
```diff
@@ -98,7 +72,37 @@ ignore = [
...
+  "E",   # pycodestyle errors
+  "F",   # Pyflakes
+  "UP",  # pyupgrade
+  "I",   # isort
+  "B",   # flake8 bugbear
+  "SIM", # flake8 simplify
+  "C90", # McCabe complexity
+#  "D",   # pydocstyle
+]
+per-file-ignores = { "tests/*" = [
```
Comment (niksirbi) [inline pyproject.toml:94]: SInce you are not currently using the "D" ruleset, you also don't need these per-file ignores

**Review by niksirbi — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (niksirbi) [reply .pre-commit-config.yaml:43]: e.g. attrs, requests, which I guess you are not using

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -6,32 +6,45 @@ repos:
...
-      rev: 24.3.0
-      hooks:
-          - id: black
+        - id: ruff-format
     - repo: https://github.com/pre-commit/mirrors-mypy
       rev: v1.9.0
       hooks:
           - id: mypy
             additional_dependencies:
+                - attrs
```
Comment (niksirbi) [inline .pre-commit-config.yaml:35]: is attrs eally necessary?

**Review by niksirbi — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 4 above)
Comment (adamltyson) [reply .pre-commit-config.yaml]: Nope :disappointed:

**Pull request merged**